src/components/views/rooms/RoomList.js

```diff
@@ -317,6 +322,99 @@ module.exports = React.createClass({
         }
     },
 
+    // Doing the stckiy headers as raw DOM, for speed, as it gets very stuttery if done
+    // properly through React
+    _initAndPositionStickyHeaders: function(initialise, scrollToPosition) {
+        var scrollArea = this._getScrollNode();
+        // Use the offset of the top of the scroll area from the window
+        // as this is used to calculate the CSS fixed top position for the stickies
+        var scrollAreaOffset = scrollArea.getBoundingClientRect().top;
+        // Use the offset of the top of the componet from the window
+        // as this is used to calculate the CSS fixed top position for the stickies
+        var scrollAreaHeight = ReactDOM.findDOMNode(this).getBoundingClientRect().height;
+
+        if (initialise) {
+            // Get a collection of sticky header containers references
+            this.stickies = document.getElementsByClassName("mx_RoomSubList_labelContainer");
+
+            // Make sure there is sufficient space to do sticky headers
+            this.scrollAreaSufficient = (120 + (this.stickies[0].getBoundingClientRect().height * this.stickies.length)) < scrollAreaHeight;
```

Comment:
Added further explanation that sticky headers requires 120 pixels plus all the sticky headers